tests/integration/targets/locale_gen/tasks/non_debian.yml

```diff
@@ -0,0 +1,44 @@
+---
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: "Non dnf RedHat system are not supported"
+  ansible.builtin.meta: "end_play"
+  when:
+    - ansible_pkg_mgr == 'yum'
+
+- name: "Install locale packages for RedHat"
+  ansible.builtin.dnf:
+    name:
+      - "glibc-langpack-pl"
+      - "glibc-langpack-en"
+    state: "present"
+  when:
+    - "ansible_os_family == 'RedHat'"
+
+- name: "Prepare locale.gen file for RedHat"
+  ansible.builtin.template:
+    src: "locale_gen.j2"
+    dest: "/etc/locale.gen"
+    owner: "root"
+    group: "root"
+    mode: "0644"
+    force: true
+  when:
+    - "ansible_os_family == 'RedHat'"
+
+- name: "Check for locale that pack are not installed"
+  locale_gen:
+    name: "ansible.UTF-8"
+    state: "present"
+  register: "locale_gen_locale_missed_result"
+  ignore_errors: true
+
+- name: "Deploy locales"
+  locale_gen:
+    name: "{{ item['locale'] + '.' + item['charset'] }}"
+    state: "present"
+  loop: "{{ vars['locale_list_simple'] | flatten(levels=1) }}"
+  loop_control:
+    label: "{{ 'Deploy locale: ' + item['locale'] }}"
```

Comment:
Please note that this only passes on RHEL 9.5 and Fedora 41 since /etc/locale.gen has been written so that the module thinks the locales are already there. This doesn't really test whether the module works: it will fail if one of the locales is comment out, since it tries to run `locale-gen`, which isn't available on both OSes.

From various places on the net (f.ex. https://www.reddit.com/r/Fedora/comments/v3w393/how_do_i_manually_install_locales/ and https://unix.stackexchange.com/questions/140299/locale-gen-command-in-centos6) it seems that locale-gen was also never available on Fedora/RHEL. So this module likely never worked with these OSes.

Maybe we should kick out the RHEL OS family in that case? Or is there any indication that locale-gen is available on these systems somehow?

Reply:
> it seems that locale-gen was also never available on Fedora/RHEL. So this module likely never worked with these OSes.

Much easier to install package, instead to maintain the locale-gen "thing". The weight of "all languages" in [RHEL is 17Mbytes](https://dl.fedoraproject.org/pub/fedora/linux/updates/41/Everything/x86_64/Packages/g/). Single langpack ~500Kbytes. Don't know why in Debian required for a custom script, still (never used this distro)

> Maybe we should kick out the RHEL OS family in that case? Or is there any indication that locale-gen is available on these systems somehow?

If locale present in `localectl list-locales` then locale ready for use